src/lib/core/mp_datetime.c

```diff
@@ -55,7 +55,7 @@ mp_sizeof_datetime(const struct datetime *date)
 struct datetime *
 datetime_unpack(const char **data, uint32_t len, struct datetime *date)
```

Comment:
I'me very confused with that your request, as I don't know how to control access to boundary conditions without my own mutator (which I don't want to write yet here). My understanding is that it's fuzzer infrastructure responsibility to mutate input `data` stream, and eventually (given infinite time) it would hit boundary condition values. 

More efficiently  boundary values are being checked in `app-tap/datetime.test.lua`, for all fields accessible from constructors. And it look good enough for today.

Reply:
Created something as simple as that (in the datetime unit test code):
```diff
diff --git a/test/unit/datetime.c b/test/unit/datetime.c
index 1dd7c59ac..4578ae8fc 100644
--- a/test/unit/datetime.c
+++ b/test/unit/datetime.c
@@ -440,6 +440,66 @@ mp_datetime_test()
 	check_plan();
 }
 
+static void
+mp_datetime_unpack_valid_checks(void)
+{
+	/* Binary, message-pack representation of datetime
+	 * payload in MP value */
+	struct binary_datetime {
+		/** Seconds since Epoch. */
+		int64_t epoch;
+		/** Nanoseconds, if any. */
+		int32_t nsec;
+		/** Offset in minutes from UTC. */
+		int16_t tzoffset;
+		/** Olson timezone id */
+		int16_t tzindex;
+	};
+
+	static struct binary_datetime invalid_values[] = {
+		{.epoch = MAX_EPOCH_SECS_VALUE + 1},
+		{.epoch = MIN_EPOCH_SECS_VALUE - 1},
+		{.nsec = MAX_NANOS_PER_SEC},
+		{.nsec = -1},
+		{.tzoffset = MIN_TZOFFSET - 1},
+		{.tzoffset = MAX_TZOFFSET + 1},
+		{.tzindex = MAX_TZINDEX + 1},
+		{.tzindex = -1},
+	};
+
+	static struct binary_datetime valid_values[] = {
+		{.epoch = MAX_EPOCH_SECS_VALUE},
+		{.epoch = MIN_EPOCH_SECS_VALUE},
+		{.nsec = MAX_NANOS_PER_SEC - 1},
+		{.nsec = 0},
+		{.tzoffset = MIN_TZOFFSET},
+		{.tzoffset = MAX_TZOFFSET},
+		{.tzindex = MAX_TZINDEX},
+		{.tzindex = 0},
+	};
+	size_t index;
+	const char* p;
+	struct datetime date;
+
+	plan(24);
+	for (index = 0; index < lengthof(valid_values); index++) {
+		struct binary_datetime value = valid_values[index];
+		p = (char*)&value;
+		memset(&date, 0, sizeof(date));
+		struct datetime *dt = datetime_unpack(&p, sizeof(value), &date);
+		isnt(dt, NULL, "datetime_unpack() is not NULL");
+		is((int64_t)dt->epoch, value.epoch, "epoch value expected");
+	}
+
+	for (index = 0; index < lengthof(valid_values); index++) {
+		struct binary_datetime value = invalid_values[index];
+		p = (char*)&value;
+		memset(&date, 0, sizeof(date));
+		struct datetime *dt = datetime_unpack(&p, sizeof(value), &date);
+		is(dt, NULL, "datetime_unpack() is NULL");
+	}
+	check_plan();
+}
 
 static int
 mp_fprint_ext_test(FILE *file, const char **data, int depth)
@@ -505,10 +565,11 @@ mp_print_test(void)
 int
 main(void)
 {
-	plan(5);
+	plan(6);
 	datetime_test();
 	tostring_datetime_test();
 	parse_date_test();
+	mp_datetime_unpack_valid_checks();
 	mp_datetime_test();
 	mp_print_test();
```